src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -156,15 +175,18 @@ static bool eliminate_reshape_v1(const shared_ptr<Node>& node) {
         if (input_node->get_output_target_inputs(0).size() != 1)
             return false;
 
-        auto shape = node->get_output_shape(0);
+        auto shape = node->get_output_partial_shape(0);
 
         // remove interchangeable nodes
-        if (input_node->get_input_partial_shape(0).is_static() && input_node->get_input_shape(0) == shape) {
+        if (input_node->get_input_partial_shape(0).is_static() && input_node->get_input_shape(0) == node->get_output_shape(0)) {
```

Comment:
node->get_output_shape(0) will throw an exception if the output shape has dyn dims

Reply:
Fixed.